### ISSUE TITLE
Why is ReUse for? DDPG

### ISSUE THREAD

**Issue by jandevries123** [MorvanZhou/Reinforcement-learning-with-tensorflow#120]
Hi in your ddpg implementation you use:

`    def _build_a(self, s, reuse=None, custom_getter=None):
        trainable = True if reuse is None else False

however is it required to have an if statement for it? because you call it here both with reuse = true

`        a_ = self._build_a(self.S_, reuse=True, custom_getter=ema_getter)   # replaced target parameters
        q_ = self._build_c(self.S_, a_, reuse=True, custom_getter=ema_getter)`

only not here:

`        self.a = self._build_a(self.S,)
        q = self._build_c(self.S, self.a, ),

` but is there a reason for that? 

I am asking this because i am trying to impement batch normalziation and i am getting a shape error with:

`            net = tf.layers.dense(s, 64, activation=tf.nn.leaky_relu, name='l1', trainable=trainable)
            print(net.get_shape())
            normie = tf.layers.BatchNormalization()(net, trainable)`

ValueError about nonetype, probably coming from the reuse being none, so i am looking to fix it by maybe removing it if not needed?

Jan
